sdk/storage/Azure.Storage.Blobs/tests/BlobClientTests.cs

```diff
@@ -443,94 +443,91 @@ async Task Verify(Task<Response<BlobContentInfo>> upload)
             long singleBlockThreshold,
             StorageTransferOptions transferOptions)
         {
-            var data = GetRandomBuffer(size);
+            using Stream stream = await CreateStream(size);
             await using DisposingContainer test = await GetTestContainerAsync();
 
             var name = GetNewBlobName();
             BlobClient blob = InstrumentClient(test.Container.GetBlobClient(name));
             var credential = new StorageSharedKeyCredential(TestConfigDefault.AccountName, TestConfigDefault.AccountKey);
             blob = InstrumentClient(new BlobClient(blob.Uri, credential, GetOptions(true)));
 
-            using (var stream = new MemoryStream(data))
+            await blob.StagedUploadAsync(
+                content: stream,
+                blobHttpHeaders: default,
+                metadata: default,
+                conditions: default,
+                progressHandler: default,
+                singleUploadThreshold: singleBlockThreshold,
+                transferOptions: transferOptions,
+                async: true);
+
+            await DownloadAndAssert(size, stream, blob);
+        }
+
+        private async Task<Stream> CreateStream(long size)
```

Comment:
`CreateStream` is such a generic looking name.  Maybe `CreateRandomDataStream` or similar would help make usage clearer?

Reply:
Ha, I went back and forth on this and was thinking CreateRandomStream or similar would be sort of unnecessary since all of the streams we are use in the tests are filled with random data. I guess a better name might be CreateLimitedMemoryStream